go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/sigstore/rekor
 
-go 1.23.4
+go 1.22.0
+
+toolchain go1.23.4
```

Comment:
this will make/force to download Go1.23.4 and then you will not build with go1.22 anymore

so i think here we need to be in go1.23.4 and the toolchain request that

Reply:
I think you're possibly misunderstanding here? The `go` directive specifies the api compatibility of the module and the `toolchain` directive specifies a suggested/recommended toolchain to use when building. Only the former has an effect on Go code that imports rekor as a module. The latter only applies as a hint to the rekor build itself and even then it is advisory and can be superseded in various ways.

Reply:
@dnwe thanks for the clarification but my understanding was that it will require updating go when building, and if any downstream projects import rekor they will be required to use the go 1.23 as well and not 1.22

But I think I did not understand that. I think I am totally wrong. This isn't very clear to me.

Reply:
Yeah for anything importing rekor toolchain is completely ignored, that’s only used if people go install rekor standalone or clone and build the repo (and even then only if their GOTOOLCHAIN contains auto)

Reply:
From what I've learned, the motivation to set `toolchain` is so that local development matches what we build the release binaries with, given we're keeping the Dockerfile Go container up to date (to pull in the latest security patches), and we'll keep `toolchain` in sync.

We should set the go.mod version to be as low as possible, based on if we need features from a newer version of Go. This is for libraries that depend on Rekor, which they themselves can choose to build with either the newest Go binary or one that matches the go.mod.

Reply:
+1 toolchain does not cascade to imports, go directive can stay 1.22 alongside using a newer toolchain to build.